File: dsa-cpp-python/Basics/Binomial-coefficient/binomial_coefficient.py
def factorial(N):
    if N == 0:
        return 1
    else:
        return N * factorial(N - 1)


n = int(input("Enter the value of n: "))
r = int(input("Enter the value of r: "))

if n < 0 or r < 0 or r > n:
    print("Invalid input")
else:
    nCr = factorial(n) // (factorial(r) * factorial(n - r))
    print(f"nCr = {nCr}")

File: Basics/Binomial-coefficient/binomial_coefficient.cpp
#include<iostream>
using namespace std;

int factorial(int n){
    if (n == 0){
        return 1;
    } 
    
    return n * factorial(n - 1);
}

int main(){
    int n, r;
    cout<<"Enter the  value of n: ";
    cin>> n;

    cout<<"Enter the  value of r: ";
    cin>> r;

    if (n < 0 or r < 0 or r > n){
        cout<< "Invalid input";
    } else{
        cout << "nCr = " << factorial(n) / (factorial(r) * factorial(n-r));
    }
    
    return 0;
}

